src/vs/workbench/api/browser/mainThreadSemanticSimilarity.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, DisposableMap } from 'vs/base/common/lifecycle';
+import { ExtHostContext, ExtHostSemanticSimilarityShape, MainContext, MainThreadSemanticSimilarityShape } from 'vs/workbench/api/common/extHost.protocol';
+import { IExtHostContext, extHostNamedCustomer } from 'vs/workbench/services/extensions/common/extHostCustomers';
+import { ISemanticSimilarityProvider, ISemanticSimilarityService } from 'vs/workbench/services/semanticSimilarity/common/semanticSimilarityService';
+
+@extHostNamedCustomer(MainContext.MainThreadSemanticSimilarity)
+export class MainThreadSemanticSimilarity extends Disposable implements MainThreadSemanticSimilarityShape {
+	private readonly _proxy: ExtHostSemanticSimilarityShape;
+	private readonly _registrations = this._register(new DisposableMap<number>());
+
+	constructor(
+		context: IExtHostContext,
+		@ISemanticSimilarityService private readonly _semanticSimilarityService: ISemanticSimilarityService
+	) {
+		super();
+		this._proxy = context.getProxy(ExtHostContext.ExtHostSemanticSimilarity);
+	}
+
+	$registerSemanticSimilarityProvider(handle: number, extensionId: string): void {
```

Comment:
Seems like extensionId isn't used? We usually don't track that along with registrations unless you just want it for logging or something

Reply:
Yeah I was going to use it for telemetry but I can remove it for now.

Reply:
removed.